File: Example/InfiniteScrollingExample/SecondViewController.m
//
//  SecondViewController.m
//  CCInfiniteScrollingExample
//
//  Created by ziryanov on 02/06/14.
//  Copyright (c) 2014 ziryanov. All rights reserved.
//

#import "SecondViewController.h"

@interface SecondViewController ()

@property (nonatomic) NSUInteger numOfCells;

@end

@implementation SecondViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    _numOfCells = 30;
    
    _tableView.estimatedRowHeight = _tableView.rowHeight;
    [_tableView setInfinityScrollingTriggerOffset:500];
    __weak SecondViewController *wself = self;
    [_tableView addBottomInfiniteScrollingWithActionHandler:^{
        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(2. * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
            wself.numOfCells += 30;
            if (wself.numOfCells >= 90)
                wself.tableView.infiniteScrollingDisabled = YES;
            [wself.tableView reloadData];
        });
    }];
}

- (IBAction)refreshPressed
{
    _numOfCells = 30;
    _tableView.infiniteScrollingDisabled = NO;
    [_tableView reloadData];
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
{
    return _tableView.rowHeight;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return _numOfCells;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"Cell"];
    cell.textLabel.text = [NSString stringWithFormat:@"cell %d", (int)indexPath.row + 1];
    return cell;
}

@end

File: SwiftExample/InfiniteScrollingExample/SecondViewController.swift
//
//  SecondViewController.swift
//  InfiniteScrollingExample
//
//  Created by ziryanov on 14/09/16.
//  Copyright © 2016 Cloud Castle. All rights reserved.
//

import UIKit
import CCInfiniteScrolling

final class SecondViewController: UIViewController, UITableViewDataSource {
    
    @IBOutlet private var tableView: UITableView!
    var numOfCells = 30

    override func viewDidLoad() {
        super.viewDidLoad()

        tableView.addBottomInfiniteScrollingWithActionHandler { [unowned self] in
            delay(1) {
                self.numOfCells += 20
                if self.numOfCells >= 90 {
                    self.tableView.infiniteScrollingDisabled = true
                }
                self.tableView.reloadData()
            }
        }
    }

    @IBAction func refreshPressed() {
        numOfCells = 20
        tableView.infiniteScrollingDisabled = false
        tableView.reloadData()
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return numOfCells
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier("Cell", forIndexPath: indexPath)
        cell.textLabel?.text = "cell \(indexPath.row + 1)"
        return cell
    }
}
